.git_hooks/pre-push

```diff
@@ -1,4 +1,11 @@
 #!/usr/bin/env sh
+fd_hard_limit="$(ulimit -n -H)"
+if [ "$fd_hard_limit" = "unlimited" ] ; then
+  ulimit -n 4096
+else
+  ulimit -n "$fd_hard_limit"
```

Comment:
Are you sure this is the right thing to do? On my machine `ulimit -n -H` gives `1048576` which seems _way_ too high.

Reply:
Given that the soft limits are just a recommendation and basically anyone can ignore them and use whatever the hard limits allow, I don't see it as a big issue.

But I don't mind capping it at 4096 if you prefer it this way.

Reply:
I prefer that I think. We have a limit of 8096 in production so if this fails with 4096 that's a canary that something's going wrong and lets us fix it before we have an outage.